OKTA-581814 Update to Event Hook rate limits (#3992)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/hooks-best-practices/index.md

* Respond immediately to the HTTP request with either a 200 (Success) or 204 (Success no content) return code.
* Process the event hook request data after sending the response code.

> A timeout of three seconds is enforced on all outbound requests for event and inline hooks. One request retry is sent in the event of a timeout or an error response from the external service. If a successful response is not received after that, a 400 error is returned with more information about what failed.

Reviewer: ```suggestion
A timeout of three seconds is enforced on all outbound requests for event and inline hooks. One request retry is sent in the event of a timeout or an error response from the external service. If a successful response isn't received after that, an HTTP 400 error is returned with more information about the failure.
```

See inline hook [Timeout and retry](/docs/concepts/inline-hooks/#timeout-and-retry) and event hook [Time out and retry](/docs/concepts/event-hooks/#timeout-and-retry).

## Limits, duplicates, and order of hook calls

The number of hook calls and the limits per org are available in the following table. Keep in mind these numbers and limits when designing your hook solution.

> Your external service that processes hook requests must consider the order of event or inline hook calls isn't guaranteed. Also, to avoid processing duplicate requests, use the `eventId` property to identify unique requests.

Reviewer: ```suggestion
Your external service that processes hook requests must consider that the order of events or inline hook calls aren't guaranteed. Also, to avoid processing duplicate requests, use the `eventId` property to identify unique requests.
```

| Hook Type | Limit Type | Limit | Description |
| --------- | -----------| ----- | ----------- |
| Event hook | Number of daily events | 400,000 | A maximum of 400,000 applicable events, per org, per day, that trigger event hooks. Event hooks aren't recorded or replayed after this point. If a request times out after three seconds, event hooks are retried once. Retries don't count toward the org limit.
|            | Maximum number of event hooks per org | 10 | A maximum of 10 active event hooks can be configured per org. You can configure each event hook to deliver multiple event types. |

> | Inline hook | Timeout | 3 seconds | inline hooks have a completion timeout of three seconds with a single retry. However, a request isn't retried if your endpoint returns a 4xx HTTP error code. Any 2xx code is considered successful, and the request is not retried. If the external service endpoint responds with a redirect, it isn't followed. |

Reviewer: ```suggestion
| Inline hook | Timeout | 3 seconds | Inline hooks have a completion timeout of three seconds with a single retry. However, a request isn't retried if your endpoint returns a 4xx HTTP error code. Any 2xx code is considered successful, and the request is not retried. If the external service endpoint responds with a redirect, it isn't followed. |
```

|             | Maximum number of inline hooks per org | 50 | The maximum number of inline hooks that can be configured per org is 50, which is a combined total for any combination of inline hook types. |
|             | Concurrent rate limit | Variable | The maximum number of inline hooks that can be sent concurrently based on org type. See [Concurrent rate limits](/docs/reference/rl-additional-limits/#concurrent-rate-limits).|

## Troubleshoot your hook implementations

Developers and administrators can preview sample Okta calls and responses from your external service. Previews are available for all event hooks ([Event hook preview](https://help.okta.com/okta_help.htm?id=ext-event-hooks-preview)) and for SAML, telephony, token, and registration inline hooks ([Preview an inline hook](https://help.okta.com/okta_help.htm?type=oie&id=ext-preview-inline-hooks)).

> Review the Admin Console System Log to troubleshoot your implementations, in addition to your external service's logging features. See the following sections to help with troubleshooting your implementations:

Reviewer: ```suggestion
Review the Admin Console System Log to troubleshoot your implementations, in addition to the logging features in your external service. See the following sections to help you with troubleshooting your implementations:
```
